storage/replica.go

```diff
@@ -1122,6 +1122,17 @@ func (r *Replica) Send(
 	return br, pErr
 }
 
+func (r *Replica) checkBatchRange(ba roachpb.BatchRequest) error {
+	rspan, err := keys.Range(ba)
+	if err != nil {
+		return err
+	}
+	return r.checkCmdHeader(roachpb.Span{
+		Key:    rspan.Key.AsRawKey(),
```

Comment:
String pulled, though I still have to use `AsRawKey` when constructing `NewRangeKeyMismatchError`. PTAL.